util/file.go

```diff
@@ -582,13 +582,39 @@ func (err PathIsNotFile) Error() string {
 // Terraform 0.14 now generates a lock file when you run `terraform init`.
 // If any such file exists, this function will copy the lock file to the destination folder
 func CopyLockFile(sourceFolder string, destinationFolder string, logger *logrus.Entry) error {
-	sourceLockFilePath := JoinPath(sourceFolder, TerraformLockFile)
-	destinationLockFilePath := JoinPath(destinationFolder, TerraformLockFile)
+	sourceLockFilePath, sourceErr := filepath.Abs(JoinPath(sourceFolder, TerraformLockFile))
+	if sourceErr != nil {
+		return errors.WithStackTrace(sourceErr)
+	}
+	destinationLockFilePath, destErr := filepath.Abs(JoinPath(destinationFolder, TerraformLockFile))
+	if destErr != nil {
+		return errors.WithStackTrace(destErr)
+	}
 
-	if FileExists(sourceLockFilePath) {
-		logger.Debugf("Copying lock file from %s to %s", sourceLockFilePath, destinationFolder)
-		return CopyFile(sourceLockFilePath, destinationLockFilePath)
+	if sourceLockFilePath == destinationLockFilePath {
+		logger.Debugf("Source and destination lock file paths are the same: %s. Not copying.", sourceLockFilePath)
+		return nil
 	}
 
-	return nil
+	if !FileExists(sourceLockFilePath) {
+		logger.Debugf("Source lock file does not exist: %s. Not copying.", sourceLockFilePath)
+		return nil
+	}
+
+	sourceContents, sourceReadErr := os.ReadFile(sourceLockFilePath)
+	if sourceReadErr != nil {
+		return errors.WithStackTrace(sourceReadErr)
+	}
+	destinationContents, destReadErr := os.ReadFile(destinationLockFilePath)
+	if destReadErr != nil {
+		return errors.WithStackTrace(destReadErr)
+	}
+
+	if string(sourceContents) == string(destinationContents) {
```

Comment:
```
util/file.go:613:5: stringXbytes: suggestion: bytes.Equal(sourceContents, destinationContents) (gocritic)
        if string(sourceContents) == string(destinationContents) {
```

Reply:
Resolved in https://github.com/gruntwork-io/terragrunt/pull/2823/commits/b3b92871b3eeeedf48a95a8a4e78f36117a5dcf9